pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -196,15 +196,11 @@ func (c *controller) onEvent(item types.NamespacedName) error {
 		Name:             item.Name + "-" + "virtualservice",
 		Namespace:        item.Namespace,
 		GroupVersionKind: gvk.VirtualService,
-		// Set this label so that we do not compare configs and just push.
-		Labels: map[string]string{constants.AlwaysPushLabel: "true"},
```

Comment:
Why is this not needed? It seems it was added for a reason and documented why; this seems to break that?

Reply:
If you look at the `needsPush`, for any non istio config, we return true.

Reply:
this is a virtual service, isn't that istiod config?

Reply:
Oh, my mistake. Just you can search this label, not read at any place except setting it.

Reply:
Originally i did remember it was useful. not sure from when it was not

Reply:
Do you still remember why do we need to always push when ingress update?

Reply:
@howardjohn By reading the ctx, i know the reason. Will fix ingress handler